compiler/AST/AstToText.cpp

```diff
@@ -0,0 +1,1318 @@
+    retval = 1;
+
+  else
+    retval = 0;
+
+  return retval;
+}
+
+// The one-based index of the first user-facing formal -> [1 .. 3]
+int AstToText::indexOfFirstFormal(FnSymbol* fn) const
+{
+  return indexForThis(fn) + 1;
+}
+
+ArgSymbol* AstToText::formalGet(FnSymbol* fn, int oneBasedIndex) const
+{
+  int      effIndex = indexForThis(fn) + oneBasedIndex;
+  DefExpr* expr     = toDefExpr(fn->formals.get(effIndex));
+
+  return toArgSymbol(expr->sym);
+}
+
+/************************************ | *************************************
+*                                                                           *
+* Normalized version for the primitive expressions found in formals.        *
+*                                                                           *
+************************************* | ************************************/
+
+void AstToText::appendExpr(Expr* expr)
+{
+  if      (UnresolvedSymExpr* sel = toUnresolvedSymExpr(expr))
+    appendExpr(sel);
+
+  else if (SymExpr*           sel = toSymExpr(expr))
+    appendExpr(sel, false);
+
+  else if (CallExpr*          sel = toCallExpr(expr))
+    appendExpr(sel);
+
+  else if (DefExpr*           sel = toDefExpr(expr))
+    appendExpr(sel);
+
+  else if (NamedExpr*         sel = toNamedExpr(expr))
+    appendExpr(sel);
+
+  else
+  {
+    // NOAKES 2015/02/05  Debugging support.
+    // Might become ASSERT in the future
+    mText += "AppendExpr.01";
+  }
+}
+
+void AstToText::appendExpr(UnresolvedSymExpr* expr)
+{
+  appendExpr(expr->unresolved);
+}
+
+void AstToText::appendExpr(SymExpr* expr, bool quoteStrings)
+{
+  if (VarSymbol* var = toVarSymbol(expr->var))
+  {
+    if (var->immediate != 0)
+    {
+      const size_t bufSize = 128;
+      char         imm[bufSize];
+
+      if (var->type == dtBool)
+      {
+        if (var->immediate->v_bool == 0)
+          sprintf(imm, "false");
+        else
+          sprintf(imm, "true");
+      }
+
+      else if (var->immediate->const_kind == CONST_KIND_STRING)
+      {
+        if (strcmp(var->immediate->v_string, "_dom") == 0)
+          strcpy(imm, "domain");
+
+        else if (quoteStrings == true)
+        {
+          char* ptr = imm;
+
+          *ptr++ = '"';
+          strcpy(ptr, var->immediate->v_string);
+          ptr = strchr(ptr, '\0');
+          *ptr++ = '"';
+          *ptr++ = '\0';
+        }
+
+        else
+          strcpy(imm, var->immediate->v_string);
+      }
+
+      else
+      {
+        snprint_imm(imm, bufSize, *var->immediate);
+
+        if (var->type != 0 && is_imag_type(var->type) == true)
+        {
+          char* tail = strchr(imm, '\0');
+
+          *tail++ = 'i';
+          *tail   = '\0';
+        }
+      }
+
+      mText += imm;
+    }
+    else
+    {
+      if (strcmp(var->name, "nil") != 0)
+        mText += var->name;
+    }
+  }
+
+  else if (ArgSymbol*  sym = toArgSymbol(expr->var))
+  {
+    mText += sym->name;
```

Comment:
You add the sym->name here, but call appendExpr(sym->name) in the next line.  Should these behave the same?